src/main/java/sysc4806/graduateAdmissions/controller/RoleController.java

```diff
@@ -42,11 +42,10 @@ public ResponseEntity getRole(@RequestParam(required=false) String name) {
      * @param role the Role to be added to the system
      * @return ResponseEntity describing the outcome of the operation
      */
-    @ResponseBody
-    @PostMapping("create")
-    public ResponseEntity createRole(@RequestBody() Role role) {
+    @PostMapping
+    public ResponseEntity createRole(@Valid @RequestBody() Role role) {
         repository.save(role);
-        return ResponseEntity.ok("Role added");
+        return ResponseEntity.status(HttpStatus.CREATED).body("Role added");
```

Comment:
I know I wasnt re-requested for a review, but since there has been a change, I thought I would point out that this will return 201 CREATED and do an update instead since there's no check if it's already in the db.